WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductRowViewModelTests.swift

```diff
@@ -426,10 +426,193 @@ final class ProductRowViewModelTests: XCTestCase {
             XCTAssertFalse(viewModel.isConfigurable)
         }
     }
+
+    // MARK: - `productSubscriptionDetails`
+    //
+    func test_productRow_when_product_type_is_subscription_and_contains_subscription_metadata_then_productRow_has_subscription_metadata() {
+        // Given
+        let rowID = Int64(0)
+        let subs: ProductSubscription = createFakeSubscription()
+        let productTypeKey = "subscription"
+
+        let product = Product.fake().copy(productID: 12,
+                                          name: "A subscription product",
+                                          productTypeKey: productTypeKey,
+                                          subscription: subs)
+
+        // When
+        let viewModel = ProductRowViewModel(id: rowID, product: product, productSubscriptionDetails: subs)
+
+        // Then
+        XCTAssertEqual(viewModel.id, rowID)
+        XCTAssertEqual(viewModel.productOrVariationID, product.productID)
+        XCTAssertEqual(viewModel.name, product.name)
+        XCTAssertNotNil(viewModel.productSubscriptionDetails)
```

Comment:
Nit: instead of checking for non-nil details, should we compare some details to make sure that the subscription details are assigned correctly?

Reply:
Initially I didn't do it because we only care for the existence of the object in this test, the properties are already tested somewhere else, but might make a better test to compare the details indeed. Updated on 48f82173d4a186055f9a05467d3ec6b13a630ff9